### PR TITLE
fix: Promote RS0030 (banned API) diagnostic to error

### LINKED ISSUE #7106: Dependency property is using `PropertyMetadata` instead of `FrameworkPropertyMetadata`
### Current behavior

https://github.com/unoplatform/uno/blob/4a2e08da0389db122ff47df87b34820e2619d57f/src/Uno.UI/UI/Xaml/FrameworkElement.Interface.skia.cs#L89

### Expected behavior

BannedApiAnalyzers should flag this.

### How to reproduce it (as minimally and precisely as possible)

_No response_

### Workaround

_No response_

### Works on UWP/WinUI

_No response_

### Environment

_No response_

### NuGet package version(s)

_No response_

### Affected platforms

_No response_

### IDE

_No response_

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): closes #7106

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Review by MartinZikmund — APPROVED** [review]

**Comment by MartinZikmund** [comment]
\azp run

**Comment by Youssef1313** [comment]
@MartinZikmund It's not yet ready. I want to see whether CI will flag the current violation as an error :)
I don't know why CI doesn't want to run 😄

**Comment by Youssef1313** [comment]
@jeromelaban @MartinZikmund Can you check why CI isn't running here?

**Comment by Youssef1313** [comment]
Finally figured out what's wrong! It's building now 🎉

File: .editorconfig [hunk 1]
```diff
@@ -222,3 +222,7 @@ dotnet_style_qualification_for_method = false:suggestion
 
 # http://kent-boogaart.com/blog/editorconfig-reference-for-c-developers#dotnet_style_qualification_for_property
 dotnet_style_qualification_for_property = false:suggestion
+
+# RS0030: Do not used banned APIs
+# https://github.com/dotnet/roslyn-analyzers/blob/main/src/Microsoft.CodeAnalysis.BannedApiAnalyzers/BannedApiAnalyzers.Help.md
+dotnet_diagnostic.RS0030.severity = error
```
Comment (Youssef1313) [inline .editorconfig:228]: There was violations that got reported as "error" in CI without this editorconfig option. So I'm assuming treat warnings as errors isn't always enabled. @jeromelaban Is that a known thing?

Comment (jeromelaban) [reply .editorconfig:228]: It's not always enabled, yes. Not for all targets at least. We're changing it gradually.

**Review by jeromelaban — APPROVED** [review]

**Comment by Youssef1313** [comment]
@jeromelaban @MartinZikmund Can you rerun the failing test? Thanks!

**Comment by Youssef1313** [comment]
Closing and reopening for new build.

**Comment by Youssef1313** [comment]
> Downloading: https://dist.nuget.org/win-x86-commandline/v4.9.1/nuget.exe
> ERR:certificate has expired

**Pull request merged**